modules/apps/portal/portal-cluster-multiple/src/main/java/com/liferay/portal/cluster/multiple/internal/jgroups/JGroupsReceiver.java

```diff
@@ -44,6 +49,36 @@ public JGroupsReceiver(
 
 		_clusterReceiver = clusterReceiver;
 		_classLoaders = classLoaders;
+
+		_fallbackFunction = (contextName, classLoaderMap) -> {
+			int index = contextName.lastIndexOf(CharPool.UNDERLINE);
+
+			if (index == -1) {
+				return null;
+			}
+
+			String contextNamePrefix = contextName.substring(0, index + 1);
+
+			for (Map.Entry<String, ClassLoader> entry :
+					classLoaderMap.entrySet()) {
+
+				String key = entry.getKey();
+
+				if (key.startsWith(contextNamePrefix)) {
```

Comment:
@tinatian this is not right, how do you know the first match is the highest version? And looping through the entrySet is very slow, considering how many bundles we got.